transformations/abbreviation_transformation/transformation.py

```diff
@@ -0,0 +1,77 @@
+import random
+import json
+import spacy
+import os.path
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+scriptpath = os.path.dirname(__file__)
+with open(os.path.join(scriptpath, 'phrase_abbrev_dict.json'),'r') as file:
+    phrase_abbrev_dict = json.loads(file.read())
+with open(os.path.join(scriptpath, 'word_abbrev_dict.json'), 'r') as file:
+    word_abbrev_dict = json.loads(file.read())
+    
+
+def abbreviate(text, nlp, prob = 0.5, seed = 0, max_outputs = 1):
+    random.seed(seed)
+    transf = []
+    for _ in range(max_outputs):
+        trans_text = text
+        for phrase in phrase_abbrev_dict:
+            trans_text = trans_text.replace(phrase, phrase_abbrev_dict[phrase])
+        doc = nlp(trans_text).doc
+        trans = []
+        for token in doc:
+            random_num = random.random()
+            word = token.text
+            if word in word_abbrev_dict and random_num < prob:
+                trans.append(word_abbrev_dict[word])
+            else:
+                trans.append(word)
+        trans1 = " ".join([str(word) for word in trans])
+        transf.append(trans1)
+        res = tuple(transf)
+    return res
+
+
+class Abbreviate(SentenceOperation):
+    tasks = [TaskType.TEXT_TO_TEXT_GENERATION,
+            TaskType.TEXT_CLASSIFICATION]
+    languages = ["en"]
+    keywords = ["lexical", "external-knowledge-based", "highly-meaning-preserving"]
+
+    def __init__(self, prob = 0.5, seed = 0, max_outputs = 1):
+        super().__init__(seed)
+        self.prob = prob
+        self.max_outputs = max_outputs
+        self.nlp = spacy.load("en_core_web_sm")

```

Comment:
You might want to use spacy like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/d39f0e5f57a79937296351996617ac03ee80fb74/transformations/close_homophones_swap/transformation.py#L55).